libraries/chain/controller.cpp

```diff
@@ -5514,7 +5514,7 @@ bool controller::validated_block_exists(const block_id_type& id) const {
 
 std::optional<signed_block_header> controller::fetch_block_header_by_id( const block_id_type& id )const {
    auto sb_ptr = my->fork_db_fetch_block_by_id(id);
-   if( sb_ptr ) return std::optional<signed_block_header>{*static_cast<signed_block_header*>(sb_ptr.get())};
+   if( sb_ptr ) return std::optional<signed_block_header>{*static_cast<const signed_block_header*>(sb_ptr.get())};
```

Comment:
I don't know why we need this case. This seems to work fine with clang:
```suggestion
   if( sb_ptr ) return std::optional<signed_block_header>{*sb_ptr};
```

Reply:
I think it is more documentation that a slice is happening.

Reply:
To me it feels like unnecessary obfuscation. I don't think making the code unnecessarily more complex is usually a good idea.

Reply:
I think making the slice obvious is being less obfuscated🤷. Personally I prefer if the language made you explicitly say you wanted a slice.